### PR TITLE
Move multiaccount data

### PR DESCRIPTION
**Pull request by yenda**
We are moving multiaccount settings to a table with proper columns and all so that we can migrate them more easily. This is the final effort of the migration work for v1.

## Testing

This touches all the multiaccount settings, network, node config, so testing would require playing around with settings as much as possible around the app and check that everything works fine. I have already done that for the happy path and a bit more so hopefully there is no bugs to be found

### CODE REVIEW

File: src/status_im/ethereum/json_rpc.cljs [hunk 1]
```diff
@@ -100,6 +98,9 @@
    "mailservers_getChatRequestRanges" {}
    "mailservers_deleteChatRequestRange" {}})
 
+(def a (atom nil))
```
Comment (cammellos) [inline src/status_im/ethereum/json_rpc.cljs]: mmmh, looks like some testing code?

File: src/status_im/ethereum/json_rpc.cljs [hunk 2]
```diff
@@ -124,6 +133,8 @@
            (if (string/blank? response)
              (on-error {:message "Blank response"})
              (let [{:keys [error result] :as response2} (types/json->clj response)]
+               (when (= method "settings_getSettings")
```
Comment (cammellos) [inline src/status_im/ethereum/json_rpc.cljs]: same here :)

**Review by cammellos — APPROVED** [review]

**Comment by churik** [comment]
@yenda 
### Issue 1 Error when tap on Privacy and Securuty setting in Profile
Screen: https://assets.saucelabs.com/jobs/ea2a4f64e12b44d4afe3b7f6f71e157d/0022screenshot.png?auth=126ec1e8c40db7686de1b036fda2da6e
Logs: https://app.saucelabs.com/tests/ab92d0a496144c4495fc630ffe19beab?auth=d0d7fe2fdb870131239e637e342f69db#42

**Comment by yenda** [comment]
@churik can you check if the tests are related?

**Comment by churik** [comment]
@yenda yes, two failures are related to your PR.
`test_ens_in_public_and_1_1_chats`
`test_can_add_existing_ens`
Can't proceed because button doesn't redirect to next screen.
Reproducible also on IOS.
0027screenshot png (768×1280) 2019-12-20 15-04-57
And If I tap multiple times, I'll get a bunch of names.
photo_2019-12-20 15 40 01

**Comment by yenda** [comment]
@churik and now?

**Comment by churik** [comment]
All e2e issues are fixed.
Does this PR block you from further work?
If no, I'll retest all settings thoroughly on Monday.

**Comment by yenda** [comment]
I'll be on holiday but I could work on fixing eventual bugs, I'd rather not risk having to do any rebase

**Comment by churik** [comment]
### Issue 1: Assets are not loaded and "This is your signing phrase" reappears after relogin

*Steps:*
- restore account with some ETH and tokens
- open wallet
- Tap "Ok, got it" in "This is your signing phrase" popup
- Reopen app
- Login
- open wallet tab

*Expected result:*
1) can see ETH and tokens
2) no more "This is your signing phrase" popup is shown

*Actual result:*
1) only  ETH
2) "This is your signing phrase" popup is shown again

Log: [relogin.log](https://github.com/status-im/status-react/files/3994690/relogin.log)

**Comment by churik** [comment]
### Issue 2: Selected mailserver is changed after relogin

*Steps:*
- create account
- go to Profile > Sync Settings > Mailserver
- unselect Automatic Selection
- select some mailserver (i.e. ac-cn-hongkong)
- relogin and check mailserver

*Expected result:*
ac-cn-hongkong is selected

*Actual result:*
another mailserver is selected

**Comment by churik** [comment]
### Issue 3: "Cyclical structure in JSON format" at attempt to change log level
*Steps:*
- create account
- go to Profile > Advanced > Log Level
- select another log level

*Expected result:*
relogin > selected log level is set

*Actual result:*
photo_2019-12-23 12 29 04
Log: 
[log-level.log](https://github.com/status-im/status-react/files/3994776/log-level.log)

**Comment by churik** [comment]
### Issue 4: Error at attempt to enable bootnodes
*Steps:*
- create account
- go to Profile > Advanced > Bootnodes
- Enable

*Expected result:*
bootnodes toggle is enabled


*Actual result:*
photo_2019-12-23 12 37 55

Log: [bootnodes-error.log](https://github.com/status-im/status-react/files/3994777/bootnodes-error.log)

**Comment by Serhy** [comment]
### Issue 5

Account won't load after re-login persistently if to enable `Use mobile data`
Every next login there are no chats in that case, no wallet, Unknown whisper account.

Screenshot 2019-12-23 at 14 43 44


**To reproduce:** 
1) Open Status and create new account
2) Being on Chats screen enable mobile data and disable WiFi connection. (**NOTE**: `Sync using Mobile data` should appear with `Remember my choice` checked by default
3) Tap `Continue syncing`
4) Re-login to account / restart app (you may enable WiFi connection again)

OR 

alternatively, after you appear on Chats screen after creating/recover new account - navigate to `Profile` -> `Sync settings` -> `Mobile data` -> enable `Use mobile data` and disable `Ask me when on mobile network` 

Logcat filtered for Status app when login in to account after above steps: 
[Login_and_no_accounts.log](https://github.com/status-im/status-react/files/3994983/Login_and_no_accounts.log)
 
All OK with develop `nightly 2019122302` build

**Comment by churik** [comment]
### Issue 6: Can't create account with keycard

*Steps:*
- create some account with keycard
- relogin

*Expected result:*
1)account is created,  "Connected" state
2) KK login screen

*Actual result:*
1) Offline state, endless spinner on + button
photo_2019-12-23 14 33 48


2) No KK account in accounts list

Log: 
[kk-no-account.log](https://github.com/status-im/status-react/files/3995123/kk-no-account.log)

**Comment by yenda** [comment]
@churik @Serhy all issues should be fixed

**Comment by churik** [comment]
@yenda 
issues 1-4 are fixed.
**ISSUE 5** is still reproducible with same steps (see https://github.com/status-im/status-react/pull/9708#issuecomment-568466250)
Log: [no-acc.log](https://github.com/status-im/status-react/files/4010066/no-acc.log)

**ISSUE 6** now not possible to login with keycard
*Steps:*
- create some account with keycard
- relogin

*Expected result:*
 KK login screen

*Actual result:*
password login screen

Log: 
[password-login.log](https://github.com/status-im/status-react/files/4009932/password-login.log)

**Comment by churik** [comment]
@yenda 
ISSUE 5 is fixed, to test Keycard I need Android build - failed with exception.
```
16:28:33  FAILURE: Build failed with an exception.
16:28:33  
16:28:33  * What went wrong:
16:28:33  Could not determine the dependencies of task ':app:lintVitalPr'.
16:28:33  > Could not resolve all task dependencies for configuration ':app:prRuntimeClasspath'.
16:28:33     > Could not resolve com.github.****.status-keycard-java:android:3.0.1.
16:28:33       Required by:
16:28:33           project :app > project :react-native-status-keycard
16:28:33        > No cached version of com.github.****.status-keycard-java:android:3.0.1 available for offline mode.
```

**Comment by yenda** [comment]
@churik rebased but I haven't checked 5 and 6 again (should be done)

**Comment by churik** [comment]
@yenda 
### Issue 7: Account is duplicated when created on different networks

Issue is not reproducible in nightly

*Steps:*
- create multiaccount
- go to wallet and add new account
- go to profile > Advanced > change network (i.e. select "Ropsten")
- login 
- go to wallet and add new account

*Expected result:*
Account 2 is added

*Actual result:*
photo_2020-01-02 11 29 32
Duplicated account disappears after relogin

**Comment by yenda** [comment]
@churik ok it's a very tiny fix status-go was returning `last-derived-path` instead of `latest-derived-path` nothing else should be affected by the change

**Comment by churik** [comment]
### Issue 8: After switching to other network with ENS name account won't load

Issue is not reproducible in nightly.

*Steps:*
- create or restore multiaccount
- add ENS name
- go to profile > Advanced > change network (i.e. select "Ropsten")
- login 

*Expected result:*
account is loaded

*Actual result:*
no account anymore (result is the same as in https://github.com/status-im/status-react/pull/9708#issuecomment-568466250)
[no-acc-ENS.log](https://github.com/status-im/status-react/files/4015691/no-acc-ENS.log)

**Comment by churik** [comment]
### Issue 9: Can't add custom bootnode

Issue is not reproducible in nightly.

*Steps:*
- create or restore multiaccount
- Go to settings -> Advanced -> Bootnodes -> +
- Try to add custom bootnode (i.e. from https://gist.github.com/jacqueswww/3ddfc5ecfa09ecea702744ab15a2200d)

*Expected result:*
bootnode is added

*Actual result:*
bootnode is not added

**Comment by yenda** [comment]
@churik 7 8 and 9 should be fixed now

**Comment by churik** [comment]
### Issue 10: SNT balance (by default on fresh accounts) disappears in wallet after relogin

Issue is not reproducible in nightly.

*Steps:*
- create or restore multiaccount
- open wallet and check that SNT and ETH are enabled by default
- reopen app

*Expected result:*
 SNT and ETH are enabled

*Actual result:*
only ETH is enabled

Moreover, if I add after this account with SNT (i.e. watch-only) - SNT balance is not fetched until relogin

**Comment by churik** [comment]
### Issue 11: Can't switch to custom network

Issue is not reproducible in nightly.

*Steps:*
- create or restore multiaccount
- open Profile > Advanced > Network > add new (i.e. https://ropsten.infura.io/v3/f315575765b14720b32382a61a89341a)
- save it
- connect to custom network
- login

*Expected result:*
can use app

*Actual result:*
Sauce Labs | Test Details: 2020-01-02 15-32-11

* Logs / Video:*
https://app.saucelabs.com/tests/c35e5c4bc02a407d8f4c7d3faf6b60ae?auth=003655ddaca1650fd01b99c37554e3de#96

**Comment by yenda** [comment]
@churik 10 and 11 should be fixed

**Comment by churik** [comment]
All issues are fixed, retested fetching balance (fix for issue 10 can affect that)

Tested on Android 10 and IOS 13:
- account creation / restoring
- account creation in offline
- balance fetching for ordinary / kk account
- bootnodes / mailserver (+autoselection) / network/ log level / fleet switching
- pairing
- ENS names
- generating new accounts
- lock multiaccount with face ID
- profile picture changing
- contacts adding / blocking
- back up seed phrase
- white out when switching apps (disabling/enabling)
- dapps permissions
- need help section
- kk account creating
- changing PIN
- kk pairing

**Pull request merged**